### PR TITLE
Added Ubuntu support by recompiling with -fPIE

### PR DESCRIPTION
**Pull request by atavakoliyext**
In order for the generated libraries to be linkable in Ubuntu and
runnable in CentOS, they must be compiled as Position-Independent
Executables (PIE). This is done by injecting the -fPIE cflag in when
compiling ImageMagick. The updated object files in this commit were
generated with this flag added.

Additional changes were made to documentation and Vagrantfile
in order to test building in Ubuntu, linking in Ubuntu via Bazel, and
running the final binary in CentOS.

The version number referenced in the README was also corrected to match
the version downloaded/built in download-build-imagemagick.sh.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,7 +59,7 @@ bazel test imagick:go_default_test
 
 ## Testing that it works in production
 
-Linux
+Linux (CentOS/7)
```
Comment (bubg-dev) [inline README.md:62]: why is CentOS tested for production but Ubuntu for dev? does dev just mean the CDbuilders in this context?

**Review by bubg-dev — COMMENTED** [review]
Just curious about the dev vs prod distinction.

File: README.md (hunk 1 above)
Comment (atavakoliyext) [reply README.md:62]: Yes, `dev` would be either developers running Linux, or build agents. The idea is to confirm cross-compilation support by building/linking in one flavor of Linux, and running on another.

**Comment by robfig** [comment]
I was surprised to see a number of new header files for rebuilding using the same version, and that the library archive more than doubled in size. Seems like there's some other difference than a new CFLAG.. any idea what?

**Comment by atavakoliyext** [comment]
I'm not sure what caused the binary size increase, but re-running the build process from the README on the previous master (68056bf18c33278b588478afdb8f4e177c238a85) on a clean repo also resulted in a sizable increase (not quite as big as this one, but still nearly double):

```
$du -h libs/linux
 38M	libs/linux

$du -h libs/linux/*
 20M	libs/linux/libMagickCore-6.Q16.a
3.9M	libs/linux/libMagickWand-6.Q16.a
2.0M	libs/linux/libjpeg.a
1.9M	libs/linux/liblcms2.a
1.2M	libs/linux/libpng.a
1.2M	libs/linux/libpng16.a
2.2M	libs/linux/libtiff.a
 92K	libs/linux/libtiffxx.a
5.3M	libs/linux/libwebp.a
136K	libs/linux/libz.a
```

My exact steps to get the above were:

```
git checkout 68056bf
git clean -fd libs
vagrant destroy -f build
vagrant up build
vagrant scp 'build:/home/vagrant/imagemagick-build/include' libs/
vagrant scp 'build:/home/vagrant/imagemagick-build/lib/*.a' libs/linux/
```

Whatever the process was to get the previous binaries, it was not the documented one. I can follow up on that with the previous commit's author.

Regarding the headers, the above process generated the same headers. I suspect they were always generated but not checked in. I left them in as a precaution, but if they aren't needed, I'm happy to remove them.

**Comment by robfig** [comment]
Weird, I guess there might have been some setting that wasn't captured in the instructions, or some of the packages changed from under them. Anyway, was just curious. SG

**Pull request merged**